{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "from pathlib import Path\n",
    "import spatialdata_io\n",
    "import spatialdata\n",
    "import json\n",
    "import time\n",
    "import pandas as pd\n",
    "import scanpy as sc\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import concurrent.futures\n",
    "from joblib import Parallel, delayed\n",
    "\n",
    "import sys\n",
    "sys.path.extend(['../../scripts','../../scripts/xenium'])\n",
    "import coexpression\n",
    "import readwrite\n",
    "cfg = readwrite.config()\n",
    "\n",
    "def diagonal_line(ax):\n",
    "    limits = np.array([ax.get_xlim(), ax.get_ylim()])\n",
    "    min_val, max_val = np.max(limits[:, 0]), np.min(limits[:, 1])  # Get the valid range in log scale\n",
    "    ax.plot([min_val, max_val], [min_val, max_val], c='k', lw=1, alpha=0.8, zorder=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cfg paths\n",
    "xenium_dir = Path(cfg['xenium_processed_data_dir'])\n",
    "xenium_raw_data_dir = Path(cfg['xenium_raw_data_dir'])\n",
    "results_dir = Path(cfg['results_dir'])\n",
    "\n",
    "# Segmentation, mapping paths\n",
    "dir_segmentations = {\n",
    "    dir_segmentation.name: (dir_segmentation)\n",
    "    for dir_segmentation in xenium_dir.iterdir()\n",
    "}\n",
    "# dir_segmentations['default'] = xenium_raw_data_dir\n",
    "\n",
    "# Read resegmentations and RCTD\n",
    "xenium_paths = {}\n",
    "umaps = {}\n",
    "\n",
    "for segmentation in (segmentations := xenium_dir.iterdir()):\n",
    "    for condition in (conditions := segmentation.iterdir()): \n",
    "        for panel in (panels := condition.iterdir()):\n",
    "            for donor in (donors := panel.iterdir()):\n",
    "                for sample in (samples := donor.iterdir()):\n",
    "                    \n",
    "                    k = (segmentation.stem,condition.stem,panel.stem,donor.stem,sample.stem)\n",
    "                    sample_path = sample / \"normalised_results/outs\"\n",
    "                    name = '/'.join(k)\n",
    "\n",
    "                    xenium_paths[k] = sample_path\n",
    "                    # if (sample_path / 'analysis/umap/gene_expression_2_components/projection.csv').exists():\n",
    "                    #     umaps[k] = pd.read_csv(sample_path / 'analysis/umap/gene_expression_2_components/projection.csv',index_col=0)\n",
    "\n",
    "\n",
    "\n",
    "# CC, pos_rate = readwrite.read_coexpression_files(cc_paths, results_dir)\n",
    "xenium_levels = ('segmentation','condition','panel','donor','sample')\n",
    "ads = readwrite.read_xenium_samples(xenium_paths,anndata_only=True,transcripts=False,sample_name_as_key=False)\n",
    "ads = pd.Series(ads.values(),\n",
    "                index=pd.Index(ads.keys(),name = xenium_levels),\n",
    "                dtype=object).sort_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "u_condition_panel = ads.index.to_frame()[['condition','panel']].drop_duplicates().index\n",
    "\n",
    "# load probe count info\n",
    "gene_panels_info = {}\n",
    "for condition_panel in u_condition_panel:\n",
    "    condition, panel = condition_panel[1:3]\n",
    "    name = '/'.join(condition_panel)\n",
    "    p=f'/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/xenium/processed/segmentation/{name}/normalised_results/outs/gene_panel.json'\n",
    "    gene_panels_info[condition, panel] = readwrite.get_gene_panel_info(p)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Xenium n° probes vs mean expr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_segmentation = '10x_0um'\n",
    "\n",
    "# merge ads and probe info\n",
    "df = {}\n",
    "for k,ad in ads.items():\n",
    "    df_ = pd.DataFrame()\n",
    "\n",
    "    condition, panel = k[1:3]\n",
    "    df_['gene_coverage']= (gene_panels_info[condition, panel] \n",
    "        .set_index('name')['gene_coverage'] \n",
    "        .loc[ad.var_names])\n",
    "    df_['mean_counts_per_gene'] = ad.X.mean(0).A1\n",
    "    df_['name'] = ad.var_names\n",
    "\n",
    "    df[k] = df_\n",
    "\n",
    "df = pd.concat(df).reset_index()\n",
    "df.columns = xenium_levels + tuple(df.columns[len(xenium_levels):])\n",
    "df['condition_panel'] = df[['condition','panel']].agg('-'.join, axis=1)\n",
    "df['condition_panel_sample'] = df[['condition','panel','sample']].agg(' '.join, axis=1)\n",
    "df_seg = df[df['segmentation']==ref_segmentation]\n",
    "\n",
    "\n",
    "# Create a FacetGrid to make subplots for each sample\n",
    "palette = dict(zip(df_seg['condition_panel'].unique(),sns.color_palette('tab10'))) \n",
    "g = sns.FacetGrid(df_seg, col=\"condition_panel_sample\", height=4, aspect=1.2, sharex=True, sharey=True, col_wrap=5)\n",
    "g.map_dataframe(sns.boxplot, x='gene_coverage', y='mean_counts_per_gene', hue='condition_panel',palette=palette, log_scale=True)\n",
    "g.add_legend(title=\"Condition-Panel\")\n",
    "g.set_axis_labels(\"Gene Coverage\", \"Mean Counts per Gene\")\n",
    "g.set_titles(\"{col_name}\")\n",
    "# plt.subplots_adjust(top=0.85)\n",
    "# g.fig.suptitle('Mean Counts per Gene vs Gene Coverage by Sample')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## scRNA n° probes vs mean expr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "scrna_probe_path = '../../../data/markers/Chromium_Human_Transcriptome_Probe_Set_v1.1.0_GRCh38-2024-A.csv'\n",
    "scrna_gene_panel_info = pd.read_csv(scrna_probe_path,skiprows=5)\n",
    "scrna_gene_coverage = scrna_gene_panel_info['gene_name'].value_counts()\n",
    "\n",
    "# p =  '../../../data/markers/Chromium_Human_Transcriptome_Probe_Set_v1.1.0_GRCh38-2024-A.probe_metadata.tsv'\n",
    "# gene_panel_metadata = pd.read_csv(p,sep='\\t')\n",
    "\n",
    "seurat_to_h5_dir = Path(cfg['results_dir']) / 'seurat_to_h5'\n",
    "\n",
    "ads_sc = {}\n",
    "for dir in seurat_to_h5_dir.iterdir():\n",
    "    if dir.is_dir():\n",
    "        ads_sc[dir.name] = sc.read_10x_h5(dir / 'RNA_counts.h5')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merge ads and probe info\n",
    "df = {}\n",
    "for k,ad in ads_sc.items():\n",
    "    df_ = pd.DataFrame()\n",
    "    \n",
    "    genes_found = [ g for g in ad.var_names if g in scrna_gene_coverage.index ]\n",
    "    df_['gene_coverage']= scrna_gene_coverage.loc[genes_found]\n",
    "    df_['mean_counts_per_gene'] = ad[:,genes_found].X.mean(0).A1\n",
    "    df_['name'] = genes_found\n",
    "\n",
    "    df[k] = df_\n",
    "\n",
    "df = pd.concat(df).reset_index()\n",
    "df.columns = ('reference',) + tuple(df.columns[1:])\n",
    "\n",
    "# Create a FacetGrid to make subplots for each sample\n",
    "palette = dict(zip(df['reference'].unique(),sns.color_palette('tab10'))) \n",
    "g = sns.FacetGrid(df, col=\"reference\", height=4, aspect=1.2, sharex=True, sharey=True, col_wrap=3)\n",
    "g.map_dataframe(sns.boxplot, x='gene_coverage', y='mean_counts_per_gene', hue='reference',palette=palette, log_scale=True)\n",
    "g.add_legend(title=\"reference\")\n",
    "g.set_axis_labels(\"Gene Coverage\", \"Mean Counts per Gene\")\n",
    "g.set_titles(\"{col_name}\")\n",
    "plt.subplots_adjust(top=0.9)\n",
    "g.fig.suptitle('scRNA probes gene coverage')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for condition_panel in u_condition_panel:\n",
    "    condition, panel = condition_panel[1:3]\n",
    "\n",
    "    # merge ads and probe info\n",
    "    df = {}\n",
    "    for k,ad in ads_sc.items():\n",
    "        df_ = pd.DataFrame()\n",
    "        gene_coverage_ = gene_panels_info[condition,panel].set_index('name')['gene_coverage'] \n",
    "\n",
    "        genes_found = [ g for g in ad.var_names if g in gene_coverage_.index ]\n",
    "        df_['gene_coverage'] = (gene_coverage_.loc[genes_found])\n",
    "        df_['mean_counts_per_gene'] = ad[:,genes_found].X.mean(0).A1\n",
    "        # df_['name'] = genes_found\n",
    "\n",
    "        df[k] = df_\n",
    "\n",
    "    df = pd.concat(df).reset_index()\n",
    "    df.columns = ('reference',) + tuple(df.columns[1:])\n",
    "\n",
    "    # Create a FacetGrid to make subplots for each sample\n",
    "    palette = dict(zip(df['reference'].unique(),sns.color_palette('tab10'))) \n",
    "    g = sns.FacetGrid(df, col=\"reference\", height=4, aspect=1.2, sharex=True, sharey=True, col_wrap=3)\n",
    "    g.map_dataframe(sns.boxplot, x='gene_coverage', y='mean_counts_per_gene', hue='reference',palette=palette, log_scale=True)\n",
    "    g.add_legend(title=\"reference\")\n",
    "    g.set_axis_labels(\"Gene Coverage\", \"Mean Counts per Gene\")\n",
    "    g.set_titles(\"{col_name}\")\n",
    "    plt.subplots_adjust(top=0.9)\n",
    "    g.fig.suptitle(f'{condition} {panel} gene coverage')\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## scRNA vs Xenium"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_sc_means = {}\n",
    "for k_sc,ad_sc in ads_sc.items():\n",
    "    df_sc_means[k_sc] = pd.Series(ad_sc.X.mean(0).A1,index = ad_sc.var_names)\n",
    "\n",
    "# merge ads and probe info\n",
    "df = {}\n",
    "for k,ad in ads.items():\n",
    "    df_ = pd.DataFrame()\n",
    "\n",
    "    condition, panel = k[1:3]\n",
    "    df_['gene_coverage']= (gene_panels_info[condition, panel] \n",
    "        .set_index('name')['gene_coverage'] \n",
    "        .loc[ad.var_names])\n",
    "    df_['mean_counts_per_gene'] = ad.X.mean(0).A1\n",
    "    df_['name'] = ad.var_names\n",
    "\n",
    "    for k_sc,df_sc_ in df_sc_means.items():\n",
    "        genes_found = [g for g in ad.var_names if g in df_sc_.index]\n",
    "        df_[f'mean_counts_per_gene_{k_sc}'] = df_sc_.loc[genes_found]\n",
    "\n",
    "    df[k] = df_\n",
    "\n",
    "df = pd.concat(df).reset_index()\n",
    "df.columns = xenium_levels + tuple(df.columns[len(xenium_levels):])\n",
    "df['condition_panel'] = df[['condition','panel']].agg('-'.join, axis=1)\n",
    "df['condition_panel_sample'] = df[['condition','panel','sample']].agg(' '.join, axis=1)\n",
    "df['gene_coverage'] = df['gene_coverage'].astype(float)\n",
    "df_seg = df[df['segmentation']=='10x_0um']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for k_sc in ads_sc.keys():\n",
    "    g = sns.lmplot(df_seg, \n",
    "        col=\"condition_panel_sample\",  x='mean_counts_per_gene', y=f'mean_counts_per_gene_{k_sc}', \n",
    "        hue='gene_coverage', palette='viridis', \n",
    "        height=5, aspect=1.2, col_wrap=5, fit_reg=False,\n",
    "        scatter_kws=dict(s=6, alpha=.5),\n",
    "        facet_kws=dict(sharex=True, sharey=True), \n",
    "    )\n",
    "\n",
    "    axes = g.fig.axes\n",
    "    for ax in axes:\n",
    "        diagonal_line(ax)\n",
    "\n",
    "    g.set(xscale=\"log\", yscale=\"log\")\n",
    "    g.set_axis_labels(\"Mean Counts per Gene Xenium\", \"Mean Counts per Gene scRNA\")\n",
    "    g.set_titles(\"{col_name}\")\n",
    "    plt.subplots_adjust(top=0.95)\n",
    "    g.fig.suptitle(k_sc)\n",
    "    plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## scRNA vs Xenium common genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "panels_genes = [ad.var_names for ad in ads.values]\n",
    "common_genes = list(set.intersection(*map(set, panels_genes)))\n",
    "df_common_genes = df_seg[df_seg['name'].isin(common_genes)]\n",
    "\n",
    "for k_sc in ads_sc.keys():\n",
    "    g = sns.lmplot(df_common_genes, \n",
    "        col=\"condition_panel_sample\",  x='mean_counts_per_gene', y=f'mean_counts_per_gene_{k_sc}', \n",
    "        hue='gene_coverage', palette='viridis', \n",
    "        height=5, aspect=1.2, col_wrap=5, fit_reg=False,\n",
    "        scatter_kws=dict(s=20, alpha=.5),\n",
    "        facet_kws=dict(sharex=True, sharey=True), \n",
    "    )\n",
    "\n",
    "    axes = g.fig.axes\n",
    "    for ax in axes:\n",
    "        diagonal_line(ax)\n",
    "\n",
    "    g.set(xscale=\"log\", yscale=\"log\")\n",
    "    g.set_axis_labels(\"Mean Counts per Gene Xenium\", \"Mean Counts per Gene scRNA\")\n",
    "    g.set_titles(\"{col_name}\")\n",
    "    plt.subplots_adjust(top=0.95)\n",
    "    g.fig.suptitle(k_sc)\n",
    "    plt.show()\n",
    "    break"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## scRNA vs Xenium common genes except breast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "panels_genes = [ad.var_names for k,ad in ads.items() if 'breast' not in k]\n",
    "common_genes = list(set.intersection(*map(set, panels_genes)))\n",
    "df_common_genes = df_seg[df_seg['name'].isin(common_genes)]\n",
    "\n",
    "for k_sc in ads_sc.keys():\n",
    "    g = sns.lmplot(df_common_genes, \n",
    "        col=\"condition_panel_sample\",  x='mean_counts_per_gene', y=f'mean_counts_per_gene_{k_sc}', \n",
    "        hue='gene_coverage', palette='viridis', \n",
    "        height=5, aspect=1.2, col_wrap=5, fit_reg=False,\n",
    "        scatter_kws=dict(s=20, alpha=.5),\n",
    "        facet_kws=dict(sharex=True, sharey=True), \n",
    "    )\n",
    "\n",
    "    axes = g.fig.axes\n",
    "    for ax in axes:\n",
    "        diagonal_line(ax)\n",
    "\n",
    "    g.set(xscale=\"log\", yscale=\"log\")\n",
    "    g.set_axis_labels(\"Mean Counts per Gene Xenium\", \"Mean Counts per Gene scRNA\")\n",
    "    g.set_titles(\"{col_name}\")\n",
    "    plt.subplots_adjust(top=0.95)\n",
    "    g.fig.suptitle(k_sc)\n",
    "    plt.show()\n",
    "    break"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "spatial",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
