{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/jbac/miniforge3/envs/spatial/lib/python3.11/importlib/__init__.py:126: FutureWarning: The legacy Dask DataFrame implementation is deprecated and will be removed in a future version. Set the configuration option `dataframe.query-planning` to `True` or None to enable the new Dask Dataframe implementation and silence this warning.\n",
      "/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/jbac/miniforge3/envs/spatial/lib/python3.11/site-packages/cudf/utils/_ptxcompiler.py:64: UserWarning: Error getting driver and runtime versions:\n",
      "\n",
      "stdout:\n",
      "\n",
      "\n",
      "\n",
      "stderr:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"<string>\", line 4, in <module>\n",
      "  File \"/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/jbac/miniforge3/envs/spatial/lib/python3.11/site-packages/numba_cuda/numba/cuda/cudadrv/driver.py\", line 272, in __getattr__\n",
      "    raise CudaSupportError(\"Error at driver init: \\n%s:\" %\n",
      "numba.cuda.cudadrv.error.CudaSupportError: Error at driver init: \n",
      "\n",
      "CUDA driver library cannot be found.\n",
      "If you are sure that a CUDA driver is installed,\n",
      "try setting environment variable NUMBA_CUDA_DRIVER\n",
      "with the file path of the CUDA driver shared library.\n",
      ":\n",
      "\n",
      "\n",
      "Not patching Numba\n",
      "/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/jbac/miniforge3/envs/spatial/lib/python3.11/site-packages/cudf/utils/gpu_utils.py:62: UserWarning: Failed to dlopen libcuda.so.1\n",
      "/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/jbac/miniforge3/envs/spatial/lib/python3.11/site-packages/anndata/utils.py:429: FutureWarning: Importing read_text from `anndata` is deprecated. Import anndata.io.read_text instead.\n"
     ]
    }
   ],
   "source": [
    "import dask\n",
    "dask.config.set({\"dataframe.query-planning\": False})\n",
    "\n",
    "import itertools\n",
    "import scanpy as sc\n",
    "import gseapy\n",
    "import liana\n",
    "import scipy\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import sys\n",
    "import argparse\n",
    "import json\n",
    "import matplotlib.patches as mpatches\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from pathlib import Path\n",
    "\n",
    "sys.path.append(\"../../../workflow/scripts/\")\n",
    "import _utils\n",
    "import readwrite\n",
    "cfg = readwrite.config()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Compute metrics diffexpr logreg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34mINFO    \u001b[0m reading                                                                                                   \n",
      "         ..\u001b[35m/../../data/xenium/processed/segmentation/10x_mm_5um/NSCLC/5k/1GQ9/1GQ9/normalised_results/outs/\u001b[0m\u001b[95mcell_fea\u001b[0m\n",
      "         \u001b[95mture_matrix.h5\u001b[0m                                                                                            \n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/jbac/miniforge3/envs/spatial/lib/python3.11/site-packages/spatialdata/_core/spatialdata.py:185: UserWarning: The table is annotating 'cell_labels', which is not present in the SpatialData object.\n",
      "/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/jbac/miniforge3/envs/spatial/lib/python3.11/site-packages/scanpy/preprocessing/_normalization.py:234: UserWarning: Some cells have zero counts\n",
      "/tmp/ipykernel_1306739/3128845931.py:109: ImplicitModificationWarning: Setting element `.obsp['spatial_connectivities']` of view, initializing view as actual.\n",
      "... storing 'label_key' as categorical\n",
      "... storing 'feature_types' as categorical\n",
      "... storing 'genome' as categorical\n",
      "/tmp/ipykernel_1306739/3128845931.py:157: ImplicitModificationWarning: Trying to modify attribute `.obs` of view, initializing view as actual.\n",
      "... storing 'has_macrophage_neighbor_str' as categorical\n",
      "2025-03-17 15:20:55,922 [WARNING] Duplicated values found in preranked stats: 91.60% of genes\n",
      "The order of those genes will be arbitrary, which may produce unexpected results.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "monocyte macrophage\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2025-03-17 15:20:56,087 [WARNING] Duplicated values found in preranked stats: 91.60% of genes\n",
      "The order of those genes will be arbitrary, which may produce unexpected results.\n",
      "2025-03-17 15:20:56,226 [WARNING] Duplicated values found in preranked stats: 95.78% of genes\n",
      "The order of those genes will be arbitrary, which may produce unexpected results.\n"
     ]
    }
   ],
   "source": [
    "segmentation = '10x_mm_5um'\n",
    "condition = 'NSCLC'\n",
    "panel = '5k'\n",
    "donor = '1GQ9'\n",
    "sample = '1GQ9'\n",
    "level = 'Level2.1'\n",
    "k = (segmentation,condition,panel,donor,sample)\n",
    "if 'proseg' in segmentation:\n",
    "    k_dir = ('proseg',condition,panel,donor,sample)\n",
    "else:\n",
    "    k_dir = k\n",
    "name = '/'.join(k)\n",
    "name_dir = '/'.join(k_dir)\n",
    "sample_corrected_counts_path = Path(f\"../../../results/resolvi_supervised/{name}/lognorm/reference_based/matched_reference_combo/rctd_class_aware/Level2.1/mixture_k=50/num_samples=30/corrected_counts.h5\")\n",
    "sample_dir = Path(f'../../../data/xenium/processed/segmentation/{name_dir}') / 'normalised_results/outs' #'raw_results'\n",
    "sample_counts = Path(f'../../../data/xenium/processed/std_seurat_analysis/{name}/lognorm/normalised_counts/data.parquet')\n",
    "sample_idx = Path(f'../../../data/xenium/processed/std_seurat_analysis/{name}/lognorm/normalised_counts/cells.parquet')\n",
    "cell_type_labels = Path(f'../../../data/xenium/processed/cell_type_annotation/{name}/lognorm/reference_based/matched_reference_combo/rctd_class_aware/{level}/single_cell/labels.parquet')\n",
    "\n",
    "out_file_df_permutations = sample_dir / 'permutation_summary.parquet'\n",
    "out_file_df_importances = sample_dir / 'importances.parquet'\n",
    "out_file_df_diffexpr = sample_dir / 'diffexpr.parquet'\n",
    "out_file_df_markers_rank_significance_logreg = sample_dir / 'markers_rank_significance_logreg.json'\n",
    "out_file_df_markers_rank_significance_diffexpr = sample_dir / 'markers_rank_significance_diffexpr.json'\n",
    "# out_dir_liana_lrdata = sample_dir / 'liana_lrdata_folder'\n",
    "\n",
    "n_neighbors = 10\n",
    "n_permutations = 30\n",
    "n_repeats = 5\n",
    "top_n = 20\n",
    "top_n_lr = 10\n",
    "cti = \"macrophage\"\n",
    "ctj = \"malignant cell\"\n",
    "scoring = 'f1'\n",
    "markers = 'diffexpr'\n",
    "# markers = \"xenium_common_markers_file\"\n",
    "\n",
    "####\n",
    "#### READ DATA\n",
    "####\n",
    "# read raw data to get spatial coordinates\n",
    "adata = readwrite.read_xenium_sample(\n",
    "    sample_dir,\n",
    "    cells_as_circles=False,\n",
    "    cells_boundaries=False,\n",
    "    cells_boundaries_layers=False,\n",
    "    nucleus_boundaries=False,\n",
    "    cells_labels=False,\n",
    "    nucleus_labels=False,\n",
    "    transcripts=False,\n",
    "    morphology_mip=False,\n",
    "    morphology_focus=False,\n",
    "    aligned_images=False,\n",
    "    anndata=True,\n",
    ")\n",
    "if 'proseg_expected' in sample_counts.as_posix():\n",
    "    adata.obs_names = 'proseg-'+adata.obs_names.astype(str)\n",
    "\n",
    "# read corrected counts\n",
    "if sample_corrected_counts_path is not None:\n",
    "    adata_corrected_counts = sc.read_10x_h5(\n",
    "        sample_corrected_counts_path,\n",
    "    )\n",
    "\n",
    "    adata_corrected_counts.obsm[\"spatial\"] = adata[adata_corrected_counts.obs_names].obsm[\"spatial\"]\n",
    "    adata = adata_corrected_counts\n",
    "\n",
    "\n",
    "# read normalised data, filter cells\n",
    "# X_normalised = pd.read_parquet(sample_counts)\n",
    "# X_normalised.index = pd.read_parquet(sample_idx).iloc[:, 0]\n",
    "# X_normalised.columns = X_normalised.columns.str.replace('.','-')\n",
    "# adata = adata[X_normalised.index,X_normalised.columns]\n",
    "# adata.layers['X_normalised'] = X_normalised\n",
    "\n",
    "# log-normalize before DE\n",
    "sc.pp.normalize_total(adata)\n",
    "sc.pp.log1p(adata)\n",
    "\n",
    "\n",
    "# read labels\n",
    "label_key = \"label_key\"\n",
    "adata.obs[label_key] = pd.read_parquet(cell_type_labels).set_index(\"cell_id\").iloc[:, 0]\n",
    "adata = adata[adata.obs[label_key].notna()]\n",
    "\n",
    "# read markers if needed\n",
    "if markers != \"diffexpr\":\n",
    "    if markers == \"xenium_common_markers_file\":\n",
    "        level_simplified = 'Level1'\n",
    "        palette = pd.read_csv('/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/xenium/metadata/col_palette_cell_types_combo.csv')\n",
    "        cell_types_mapping = palette.set_index(level)[level_simplified].replace(r' of .+', '', regex=True)\n",
    "        cell_types_mapping[cell_types_mapping.str.contains('malignant')] = 'malignant cell'\n",
    "        adata.obs[label_key] = adata.obs[label_key].replace(cell_types_mapping)\n",
    "        df_markers = pd.read_csv('/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/markers/Xenium_panels_common_markers.csv')[[\"cell_type\",\"gene_name\"]]\n",
    "    else:\n",
    "        df_markers = pd.read_csv(markers)[[\"cell_type\",\"gene_name\"]]\n",
    "\n",
    "    ct_not_found = adata.obs[label_key][~adata.obs[label_key].isin(df_markers['cell_type'])].unique()\n",
    "    print(f\"Could not find {ct_not_found} in markers file\")\n",
    "    adata = adata[adata.obs[label_key].isin(df_markers['cell_type'])]\n",
    "\n",
    "\n",
    "# get kNN graph\n",
    "obsm = 'spatial'\n",
    "knnlabels, knndis, knnidx, knn_graph = _utils.get_knn_labels(\n",
    "    adata,n_neighbors=n_neighbors,\n",
    "    label_key=label_key,obsm=obsm,\n",
    "    return_sparse_neighbors=True)\n",
    "adata.obsp[f'{obsm}_connectivities'] = knn_graph\n",
    "\n",
    "# iterate over targets permutations (cell type i with cell type j presence in kNN)\n",
    "df_diffexpr = {}\n",
    "df_markers_rank_significance_diffexpr = {}\n",
    "u_cell_types = adata.obs[label_key].unique()\n",
    "df_ctj_marker_genes = {}\n",
    "\n",
    "\n",
    "for ctj in u_cell_types:\n",
    "    if  (adata.obs[label_key]==ctj).sum() < 30:\n",
    "        print(f\"Not enough cells from class {ctj}\")\n",
    "        continue\n",
    "\n",
    "    # get markers\n",
    "    if markers == \"diffexpr\":\n",
    "        sc.tl.rank_genes_groups(adata, groupby=label_key, groups=[ctj], reference='rest', method=\"wilcoxon\")\n",
    "        ctj_marker_genes = sc.get.rank_genes_groups_df(adata, group=ctj)['names'][: top_n].tolist()\n",
    "    else:\n",
    "        ctj_marker_genes = df_markers[df_markers[\"cell_type\"] == ctj][\"gene_name\"].tolist()\n",
    "        ctj_marker_genes = [g for g in ctj_marker_genes if g in adata.var_names]\n",
    "\n",
    "        if len(ctj_marker_genes) == 0:\n",
    "            print(f\"no markers found for {ctj}\")\n",
    "            continue\n",
    "\n",
    "    df_ctj_marker_genes[ctj] = ctj_marker_genes\n",
    "\n",
    "    for cti in u_cell_types:\n",
    "        if cti == ctj:\n",
    "            continue\n",
    "        print(cti, ctj)\n",
    "\n",
    "        adata.obs[f\"has_{ctj}_neighbor\"] = knnlabels[ctj]>0\n",
    "\n",
    "        # Filter for cti\n",
    "        adata_cti = adata[adata.obs[label_key] == cti]\n",
    "\n",
    "        if (   (adata_cti.obs[f\"has_{ctj}_neighbor\"]).sum() < 30\n",
    "            or (~adata_cti.obs[f\"has_{ctj}_neighbor\"]).sum() < 30\n",
    "            ):\n",
    "            print(f\"Not enough cells from each class to test {cti} with {ctj} neighbors\")\n",
    "            continue\n",
    "\n",
    "\n",
    "        ###\n",
    "        ### DIFF EXPR TEST: check DE genes between cti with ctj neighbor or not\n",
    "        ###\n",
    "        adata_cti.obs[f'has_{ctj}_neighbor_str'] = adata_cti.obs[f'has_{ctj}_neighbor'].astype(str)\n",
    "        sc.tl.rank_genes_groups(adata_cti, groupby=f\"has_{ctj}_neighbor_str\", groups=['True'], reference='False', method=\"wilcoxon\")\n",
    "        df_diffexpr[cti,ctj] = sc.get.rank_genes_groups_df(adata_cti, group='True').sort_values('pvals_adj')\n",
    "        df_diffexpr[cti, ctj]['-log10pvals_x_logfoldchanges'] = -np.log10(df_diffexpr[cti, ctj]['pvals']) * df_diffexpr[cti, ctj]['logfoldchanges']\n",
    "        df_diffexpr[cti, ctj]['-log10pvals_x_signFC'] = -np.log10(df_diffexpr[cti, ctj]['pvals']) * np.sign(df_diffexpr[cti, ctj]['logfoldchanges'])\n",
    "\n",
    "        # get significance from gsea and hypergeometric test\n",
    "        df_markers_rank_significance_diffexpr[cti, ctj] = pd.DataFrame()\n",
    "        for rank_metric in ['logfoldchanges', '-log10pvals_x_logfoldchanges', '-log10pvals_x_signFC']:\n",
    "            df_markers_rank_significance_diffexpr[cti, ctj][rank_metric] = _utils.get_marker_rank_significance(\n",
    "                rnk=df_diffexpr[cti, ctj].set_index(\"names\")[rank_metric].sort_values(ascending=False),\n",
    "                gene_set=ctj_marker_genes,\n",
    "                top_n=top_n,\n",
    "            ).iloc[0]\n",
    "\n",
    "        break\n",
    "    break\n",
    "\n",
    "###\n",
    "### SAVE OUTPUTS\n",
    "###\n",
    "# df_permutations_logreg = pd.concat(df_permutations_logreg)\n",
    "# df_importances_logreg = pd.concat(df_importances_logreg)\n",
    "# df_diffexpr = pd.concat(df_diffexpr)\n",
    "# df_markers_rank_significance_logreg = pd.concat(df_markers_rank_significance_logreg)\n",
    "df_markers_rank_significance_diffexpr = pd.concat(df_markers_rank_significance_diffexpr)\n",
    "\n",
    "#logreg\n",
    "# df_permutations.to_parquet(out_file_df_permutations)\n",
    "# df_importances.to_parquet(out_file_df_importances)\n",
    "# df_markers_rank_significance_logreg.to_parquet(out_file_df_markers_rank_significance_logreg)\n",
    "\n",
    "# #diffexpr\n",
    "# df_diffexpr.to_parquet(out_file_df_diffexpr)\n",
    "# df_markers_rank_significance_diffexpr.to_parquet(out_file_df_markers_rank_significance_diffexpr)\n",
    "\n",
    "#liana\n",
    "# readwrite.write_anndata_folder(lrdata, outm_dir_liana_lrdata)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Compute metrics marker purity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/jbac/miniforge3/envs/spatial/lib/python3.11/site-packages/anndata/_core/aligned_df.py:68: ImplicitModificationWarning: Transforming to str index.\n",
      "/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/jbac/miniforge3/envs/spatial/lib/python3.11/site-packages/spatialdata/_core/spatialdata.py:185: UserWarning: The table is annotating 'cell_polygons', which is not present in the SpatialData object.\n",
      "/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/jbac/miniforge3/envs/spatial/lib/python3.11/site-packages/anndata/_core/anndata.py:822: UserWarning: \n",
      "AnnData expects .obs.index to contain strings, but got values like:\n",
      "    [0, 1, 2, 3, 4]\n",
      "\n",
      "    Inferred to be: integer\n",
      "\n",
      "/tmp/ipykernel_1891581/602649072.py:69: ImplicitModificationWarning: Setting element `.layers['X_normalised']` of view, initializing view as actual.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "metrics_summary.csv not found at: /work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/xenium/processed/segmentation/proseg/NSCLC/lung/0S8R/0S8R/raw_results/metrics_summary.csv\n",
      "Could not find ['cycling lymphocyte'] in markers file\n",
      "stromal cell malignant cell\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1891581/602649072.py:79: ImplicitModificationWarning: Trying to modify attribute `.obs` of view, initializing view as actual.\n",
      "/tmp/ipykernel_1891581/602649072.py:136: ImplicitModificationWarning: Trying to modify attribute `.obs` of view, initializing view as actual.\n"
     ]
    }
   ],
   "source": [
    "segmentation = 'proseg_expected'\n",
    "condition = 'NSCLC'\n",
    "panel = 'lung'\n",
    "donor = '0S8R'\n",
    "sample = '0S8R'\n",
    "k = (segmentation,condition,panel,donor,sample)\n",
    "if 'proseg' in segmentation:\n",
    "    k_dir = ('proseg',condition,panel,donor,sample)\n",
    "else:\n",
    "    k_dir = k\n",
    "name = '/'.join(k)\n",
    "name_dir = '/'.join(k_dir)\n",
    "sample_dir = Path(f'/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/xenium/processed/segmentation/{name_dir}') / 'raw_results'\n",
    "sample_counts = Path(f'/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/xenium/processed/std_seurat_analysis/{name}/lognorm/normalised_counts/data.parquet')\n",
    "sample_idx = Path(f'/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/xenium/processed/std_seurat_analysis/{name}/lognorm/normalised_counts/cells.parquet')\n",
    "cell_type_labels = Path(f'/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/xenium/processed/cell_type_annotation/{name}/lognorm/reference_based/matched_reference_combo/rctd_class_aware/Level2.1/single_cell/labels.parquet')\n",
    "\n",
    "out_file_df_permutations = sample_dir / 'permutation_summary.parquet'\n",
    "out_file_df_importances = sample_dir / 'importances.parquet'\n",
    "out_file_df_diffexpr = sample_dir / 'diffexpr.parquet'\n",
    "out_file_df_markers_rank_significance_logreg = sample_dir / 'markers_rank_significance_logreg.json'\n",
    "out_file_df_markers_rank_significance_diffexpr = sample_dir / 'markers_rank_significance_diffexpr.json'\n",
    "# out_dir_liana_lrdata = sample_dir / 'liana_lrdata_folder'\n",
    "\n",
    "n_neighbors = 10\n",
    "n_permutations = 30\n",
    "n_repeats = 5\n",
    "top_n = 20\n",
    "top_n_lr = 10\n",
    "cti = \"macrophage\"\n",
    "ctj = \"malignant cell\"\n",
    "scoring = 'f1'\n",
    "markers = '/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/markers/Xenium_panels_common_markers.csv'\n",
    "# markers = 'diffexpr'\n",
    "\n",
    "\n",
    "level_simplified = 'Level1'\n",
    "palette = pd.read_csv('/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/xenium/metadata/col_palette_cell_types_combo.csv')\n",
    "cell_types_mapping = palette.set_index('Level2.1')[level_simplified]\n",
    "cell_types_mapping[cell_types_mapping.str.contains('malignant')] = 'malignant cell'\n",
    "\n",
    "####\n",
    "#### READ DATA\n",
    "####\n",
    "# read raw data to get spatial coordinates\n",
    "adata = readwrite.read_xenium_sample(\n",
    "    sample_dir,\n",
    "    cells_as_circles=False,\n",
    "    cells_boundaries=False,\n",
    "    cells_boundaries_layers=False,\n",
    "    nucleus_boundaries=False,\n",
    "    cells_labels=False,\n",
    "    nucleus_labels=False,\n",
    "    transcripts=False,\n",
    "    morphology_mip=False,\n",
    "    morphology_focus=False,\n",
    "    aligned_images=False,\n",
    "    anndata=True,\n",
    ")\n",
    "if 'proseg_expected' in sample_counts.as_posix():\n",
    "    adata.obs_names = 'proseg-'+adata.obs_names.astype(str)\n",
    "\n",
    "\n",
    "# read normalised data, filter cells\n",
    "X_normalised = pd.read_parquet(sample_counts)\n",
    "X_normalised.index = pd.read_parquet(sample_idx).iloc[:, 0]\n",
    "X_normalised.columns = X_normalised.columns.str.replace('.','-')\n",
    "adata = adata[X_normalised.index,X_normalised.columns]\n",
    "adata.layers['X_normalised'] = X_normalised\n",
    "\n",
    "# log-normalize before DE\n",
    "sc.pp.normalize_total(adata)\n",
    "sc.pp.log1p(adata)\n",
    "\n",
    "# read labels\n",
    "label_key = \"label_key\"\n",
    "adata.obs[label_key] = pd.read_parquet(cell_type_labels).set_index(\"cell_id\").iloc[:, 0]\n",
    "adata = adata[adata.obs[label_key].notna()]\n",
    "adata.obs[label_key] = adata.obs[label_key].replace(cell_types_mapping)\n",
    "\n",
    "# read markers if needed\n",
    "if markers != \"diffexpr\":\n",
    "    if markers == \"xenium_common_markers_file\":\n",
    "        level_simplified = 'Level1'\n",
    "        palette = pd.read_csv('/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/xenium/metadata/col_palette_cell_types_combo.csv')\n",
    "        cell_types_mapping = palette.set_index(level)[level_simplified].replace(r' of .+', '', regex=True)\n",
    "        cell_types_mapping[cell_types_mapping.str.contains('malignant')] = 'malignant cell'\n",
    "        adata.obs[label_key] = adata.obs[label_key].replace(cell_types_mapping)\n",
    "        df_markers = pd.read_csv('/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/markers/Xenium_panels_common_markers.csv')[[\"cell_type\",\"gene_name\"]]\n",
    "    else:\n",
    "        df_markers = pd.read_csv(markers)[[\"cell_type\",\"gene_name\"]]\n",
    "\n",
    "    ct_not_found = adata.obs[label_key][~adata.obs[label_key].isin(df_markers['cell_type'])].unique()\n",
    "    print(f\"Could not find {ct_not_found} in markers file\")\n",
    "    adata = adata[adata.obs[label_key].isin(df_markers['cell_type'])]\n",
    "\n",
    "\n",
    "# get kNN graph\n",
    "# obsm = 'spatial'\n",
    "# knnlabels, knndis, knnidx, knn_graph = _utils.get_knn_labels(\n",
    "#     adata,n_neighbors=n_neighbors,\n",
    "#     label_key=label_key,obsm=obsm,\n",
    "#     return_sparse_neighbors=True)\n",
    "# adata.obsp[f'{obsm}_connectivities'] = knn_graph\n",
    "\n",
    "\n",
    "# iterate over targets permutations (cell type i with cell type j presence in kNN)\n",
    "u_cell_types = adata.obs[label_key].unique()\n",
    "df_ctj_marker_genes = {}\n",
    "\n",
    "\n",
    "for ctj in u_cell_types:\n",
    "    if  (adata.obs[label_key]==ctj).sum() < 30:\n",
    "        print(f\"Not enough cells from class {ctj}\")\n",
    "        continue\n",
    "\n",
    "    # get markers\n",
    "    if markers == \"diffexpr\":\n",
    "        sc.tl.rank_genes_groups(adata, groupby=label_key, groups=[ctj], reference='rest', method=\"wilcoxon\")\n",
    "        ctj_marker_genes = sc.get.rank_genes_groups_df(adata, group=ctj)['names'][: top_n].tolist()\n",
    "    else:\n",
    "        ctj_marker_genes = df_markers[df_markers[\"cell_type\"] == ctj][\"gene_name\"].tolist()\n",
    "        ctj_marker_genes = [g for g in ctj_marker_genes if g in adata.var_names]\n",
    "\n",
    "        if len(ctj_marker_genes) == 0:\n",
    "            print(f\"no markers found for {ctj}\")\n",
    "            continue\n",
    "\n",
    "    df_ctj_marker_genes[ctj] = ctj_marker_genes\n",
    "\n",
    "    for cti in u_cell_types:\n",
    "        if cti == ctj:\n",
    "            continue\n",
    "        print(cti, ctj)\n",
    "\n",
    "        adata.obs[f\"has_{ctj}_neighbor\"] = knnlabels[ctj]>0\n",
    "\n",
    "        # Filter for cti\n",
    "        adata_cti = adata[adata.obs[label_key] == cti]\n",
    "\n",
    "        if (   (adata_cti.obs[f\"has_{ctj}_neighbor\"]).sum() < 30\n",
    "            or (~adata_cti.obs[f\"has_{ctj}_neighbor\"]).sum() < 30\n",
    "            ):\n",
    "            print(f\"Not enough cells from each class to test {cti} with {ctj} neighbors\")\n",
    "            continue\n",
    "\n",
    "\n",
    "        ###\n",
    "        ### DIFF EXPR TEST: check DE genes between cti with ctj neighbor or not\n",
    "        ###\n",
    "        # adata_cti.obs[f'has_{ctj}_neighbor_str'] = adata_cti.obs[f'has_{ctj}_neighbor'].astype(str)\n",
    "        # sc.tl.rank_genes_groups(adata_cti, groupby=f\"has_{ctj}_neighbor_str\", groups=['True'], reference='False', method=\"wilcoxon\")\n",
    "        # df_diffexpr[cti,ctj] = sc.get.rank_genes_groups_df(adata_cti, group='True').sort_values('pvals_adj')\n",
    "\n",
    "\n",
    "        # # get significance from gsea and hypergeometric test\n",
    "        # df_markers_rank_significance_diffexpr[cti, ctj] = _utils.get_marker_rank_significance(\n",
    "        #     rnk=df_diffexpr[cti, ctj].set_index(\"names\")[\"logfoldchanges\"],\n",
    "        #     gene_set=ctj_marker_genes,\n",
    "        #     top_n=top_n,\n",
    "        # )"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plot results diffexpr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cfg paths\n",
    "xenium_dir = Path(cfg['xenium_processed_data_dir'])\n",
    "xenium_std_seurat_analysis_dir = Path(cfg['xenium_std_seurat_analysis_dir'])\n",
    "xenium_cell_type_annotation_dir = Path(cfg['xenium_cell_type_annotation_dir'])\n",
    "results_dir = Path(cfg['results_dir'])\n",
    "palette_dir = Path(cfg['xenium_metadata_dir'])\n",
    "\n",
    "# Params\n",
    "# probably only need to run for lognorm data\n",
    "normalisations = ['lognorm',]\n",
    "layers = ['data',]\n",
    "reference = 'matched_reference_combo'\n",
    "method = 'rctd_class_aware'\n",
    "level = 'Level2.1'\n",
    "segmentation_palette = palette_dir / 'col_palette_segmentation.csv'\n",
    "\n",
    "n_neighbors = 10\n",
    "n_permutations = 30\n",
    "n_repeats = 5\n",
    "top_n = 20\n",
    "scoring = 'f1'\n",
    "markers = 'diffexpr' #'/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/markers/cellmarker_cell_types_markers.json'\n",
    "\n",
    "# needed to get unique cell types names for each level\n",
    "# cell_types_palette = pd.read_csv(palette_dir / 'col_palette_cell_types_combo.csv')\n",
    "\n",
    "df_diffexpr = {}\n",
    "df_markers_rank_significance_diffexpr = {}\n",
    "for segmentation in (segmentations := xenium_std_seurat_analysis_dir.iterdir()):\n",
    "    for condition in (conditions := segmentation.iterdir()): \n",
    "        for panel in (panels := condition.iterdir()):\n",
    "            for donor in (donors := panel.iterdir()):\n",
    "                for sample in (samples := donor.iterdir()):\n",
    "                    for normalisation in normalisations:\n",
    "                        for layer in layers:\n",
    "                            # for reference in references:\n",
    "                            #     for method in methods:\n",
    "                            #         for level in levels:\n",
    "\n",
    "                            k = (segmentation.stem,condition.stem,panel.stem,donor.stem,sample.stem)\n",
    "                            name = '/'.join(k)\n",
    "\n",
    "                            out_file_df_diffexpr = results_dir / f'contamination_metrics_diffexpr/{name}/{normalisation}/{layer}_{reference}_{method}_{level}_diffexpr.parquet'\n",
    "                            out_file_df_markers_rank_significance_diffexpr = results_dir / f'contamination_metrics_diffexpr/{name}/{normalisation}/{layer}_{reference}_{method}_{level}_markers_rank_significance_diffexpr.parquet'\n",
    "\n",
    "                            if out_file_df_diffexpr.exists():\n",
    "                                # df_diffexpr[k] = pd.read_parquet(out_file_df_diffexpr)\n",
    "                                df_markers_rank_significance_diffexpr[k] = pd.read_parquet(out_file_df_markers_rank_significance_diffexpr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cti = 'T cell'\n",
    "ctj = 'malignant cell'\n",
    "xenium_levels = [\"segmentation\", \"condition\", \"panel\", \"donor\", \"sample\",\"cti\",\"ctj\"]\n",
    "\n",
    "hue = \"segmentation\"\n",
    "hue_order = [\n",
    "    \"10x_mm_0um\",\n",
    "    \"10x_mm_5um\",\n",
    "    \"10x_mm_15um\",\n",
    "    \"10x_0um\",\n",
    "    \"10x_5um\",\n",
    "    \"10x_15um\",\n",
    "    \"baysor\",\n",
    "    \"proseg_expected\",\n",
    "    \"proseg_mode\",\n",
    "    \"segger\",\n",
    "]\n",
    "\n",
    "\n",
    "palette = pd.read_csv(segmentation_palette, index_col=0).iloc[:, 0]\n",
    "\n",
    "\n",
    "df = pd.concat(df_markers_rank_significance_diffexpr).reset_index()\n",
    "df.columns = xenium_levels + df.columns[len(xenium_levels) :].tolist()\n",
    "df = df.query(\"cti == @cti and ctj == @ctj\")\n",
    "df['-log10pvalue'] = -np.log10(df['hypergeometric_pvalue'])\n",
    "\n",
    "\n",
    "# plotting params, palette\n",
    "title = f\"Reference: {reference}, Method: {method}, Level: {level} \\n{cti} contaminated by {ctj}\"\n",
    "unique_labels = [c for c in hue_order if c in np.unique(df[hue].dropna())]\n",
    "unique_labels = unique_labels + [c for c in np.unique(df[hue].dropna()) if c not in unique_labels]\n",
    "palette = {u: palette[u] for u in unique_labels}\n",
    "legend_handles = [mpatches.Patch(color=color, label=label) for label, color in palette.items()]\n",
    "\n",
    "sns.set(style=\"ticks\")\n",
    "\n",
    "### hypergeometric pvalue boxplot\n",
    "f = plt.figure(figsize=(6, 6))\n",
    "ax = plt.subplot()\n",
    "g = sns.boxplot(df,x='panel',y='-log10pvalue', \n",
    "                hue=hue, hue_order=unique_labels, \n",
    "                legend=False, palette=palette,ax=ax\n",
    "                )\n",
    "\n",
    "plt.ylabel(r'$-\\log_{10} \\text{ p-value}$')\n",
    "sns.despine(offset=10, trim=True)\n",
    "ax.yaxis.grid(True)\n",
    "\n",
    "plt.suptitle(title)\n",
    "f.legend(\n",
    "    handles=legend_handles,\n",
    "    loc=\"center left\",\n",
    "    bbox_to_anchor=(1, 0.5),\n",
    "    title=hue,\n",
    "    frameon=False,\n",
    ")\n",
    "plt.tight_layout(rect=[0, 0, 1, 0.95])\n",
    "# plt.savefig(out_file, dpi=dpi, bbox_inches=\"tight\")\n",
    "plt.show()\n",
    "\n",
    "\n",
    "### NES boxplot\n",
    "f = plt.figure(figsize=(6, 6))\n",
    "ax = plt.subplot()\n",
    "g = sns.boxplot(df,x='panel',y='NES', \n",
    "                hue=hue, hue_order=unique_labels, \n",
    "                legend=False, palette=palette,ax=ax\n",
    "                )\n",
    "\n",
    "sns.despine(offset=10, trim=True)\n",
    "ax.yaxis.grid(True)\n",
    "\n",
    "plt.suptitle(title)\n",
    "f.legend(\n",
    "    handles=legend_handles,\n",
    "    loc=\"center left\",\n",
    "    bbox_to_anchor=(1, 0.5),\n",
    "    title=hue,\n",
    "    frameon=False,\n",
    ")\n",
    "plt.tight_layout(rect=[0, 0, 1, 0.95])\n",
    "# plt.savefig(out_file, dpi=dpi, bbox_inches=\"tight\")\n",
    "plt.show()\n",
    "\n",
    "\n",
    "### number of hits boxplot\n",
    "f = plt.figure(figsize=(6, 6))\n",
    "ax = plt.subplot()\n",
    "g = sns.boxplot(df,x='panel',y=f'n_hits_{top_n=}', \n",
    "                hue=hue, hue_order=unique_labels, \n",
    "                legend=False, palette=palette,ax=ax\n",
    "                )\n",
    "\n",
    "sns.despine(offset=10, trim=True)\n",
    "ax.yaxis.grid(True)\n",
    "\n",
    "plt.suptitle(title)\n",
    "f.legend(\n",
    "    handles=legend_handles,\n",
    "    loc=\"center left\",\n",
    "    bbox_to_anchor=(1, 0.5),\n",
    "    title=hue,\n",
    "    frameon=False,\n",
    ")\n",
    "plt.tight_layout(rect=[0, 0, 1, 0.95])\n",
    "# plt.savefig(out_file, dpi=dpi, bbox_inches=\"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.set_style('ticks')\n",
    "ref_segmentation = '10x_5um'\n",
    "\n",
    "df = pd.concat(df_markers_rank_significance_diffexpr).reset_index()\n",
    "df['-log10pvalue'] = -np.log10(df['hypergeometric_pvalue'])\n",
    "df.columns = xenium_levels + df.columns[len(xenium_levels) :].tolist()\n",
    "u_condition_panel = df[['condition','panel']].drop_duplicates().values\n",
    "\n",
    "metrics = ['NES', '-log10pvalue', f'n_hits_{top_n=}']\n",
    "\n",
    "for metric in metrics:\n",
    "    for condition,panel in u_condition_panel:\n",
    "\n",
    "        df_plot = df.query(f\"segmentation == '{ref_segmentation}' and condition == '{condition}' and panel == '{panel}'\")\n",
    "        df_plot = df_plot.groupby(['cti', 'ctj'])[metric].mean().unstack()\n",
    "        df_plot = df_plot.loc[df_plot.sum(1).sort_values(ascending=False).index]\n",
    "        df_plot = df_plot[df_plot.sum(0).sort_values(ascending=False).index]\n",
    "\n",
    "        f = plt.figure(figsize=(8,8))\n",
    "        ax = plt.subplot()\n",
    "        ax.set_title(f\"{condition=} {panel=} {metric=}\",fontsize=20)\n",
    "        g = sns.heatmap(df_plot,cmap='coolwarm',center=-np.log10(0.05) if metric == '-log10pvalue' else 0.)\n",
    "        plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plot results logreg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cfg paths\n",
    "xenium_dir = Path(cfg['xenium_processed_data_dir'])\n",
    "xenium_std_seurat_analysis_dir = Path(cfg['xenium_std_seurat_analysis_dir'])\n",
    "xenium_cell_type_annotation_dir = Path(cfg['xenium_cell_type_annotation_dir'])\n",
    "results_dir = Path(cfg['results_dir'])\n",
    "palette_dir = Path(cfg['xenium_metadata_dir'])\n",
    "\n",
    "# Params\n",
    "normalisations = ['lognorm',]\n",
    "layers = ['data',]\n",
    "reference = 'matched_reference_combo'\n",
    "method = 'rctd_class_aware'\n",
    "level = 'Level2.1'\n",
    "segmentation_palette = palette_dir / 'col_palette_segmentation.csv'\n",
    "\n",
    "n_neighbors = 10\n",
    "n_permutations = 30\n",
    "n_repeats = 5\n",
    "top_n = 20\n",
    "scoring = 'f1'\n",
    "markers = 'diffexpr' #'/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/markers/cellmarker_cell_types_markers.json'\n",
    "\n",
    "# needed to get unique cell types names for each level\n",
    "# cell_types_palette = pd.read_csv(palette_dir / 'col_palette_cell_types_combo.csv')\n",
    "\n",
    "df_diffexpr = {}\n",
    "df_markers_rank_significance_diffexpr = {}\n",
    "for segmentation in (segmentations := xenium_std_seurat_analysis_dir.iterdir()):\n",
    "    for condition in (conditions := segmentation.iterdir()): \n",
    "        for panel in (panels := condition.iterdir()):\n",
    "            for donor in (donors := panel.iterdir()):\n",
    "                for sample in (samples := donor.iterdir()):\n",
    "                    for normalisation in normalisations:\n",
    "                        for layer in layers:\n",
    "                            # for reference in references:\n",
    "                            #     for method in methods:\n",
    "                            #         for level in levels:\n",
    "\n",
    "                            k = (segmentation.stem,condition.stem,panel.stem,donor.stem,sample.stem)\n",
    "                            name = '/'.join(k)\n",
    "\n",
    "                            out_file_df_permutations_logreg = results_dir / f'contamination_metrics_logreg/{name}/{normalisation}/{layer}_{reference}_{method}_{level}_permutations_logreg.parquet'\n",
    "                            out_file_df_importances_logreg = results_dir / f'contamination_metrics_logreg/{name}/{normalisation}/{layer}_{reference}_{method}_{level}_importances_logreg.parquet'\n",
    "                            out_file_df_markers_rank_significance_logreg = results_dir / f'contamination_metrics_logreg/{name}/{normalisation}/{layer}_{reference}_{method}_{level}_markers_rank_significance_logreg.json'\n",
    "\n",
    "                            if out_file_df_permutations_logreg.exists():\n",
    "                                # df_diffexpr[k] = pd.read_parquet(out_file_df_diffexpr)\n",
    "                                df_permutations_logreg[k] = pd.read_parquet(out_file_df_permutations_logreg)\n",
    "                                df_importances_logreg[k] = pd.read_parquet(out_file_df_importances_logreg)\n",
    "                                df_markers_rank_significance_logreg[k] = pd.read_parquet(out_file_df_markers_rank_significance_logreg)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'df_markers_rank_significance_logreg' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[6], line 5\u001b[0m\n\u001b[1;32m      2\u001b[0m ctj \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mmalignant cell\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m      3\u001b[0m xenium_levels \u001b[38;5;241m=\u001b[39m [\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124msegmentation\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcondition\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mpanel\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdonor\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124msample\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcti\u001b[39m\u001b[38;5;124m\"\u001b[39m,\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mctj\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[0;32m----> 5\u001b[0m df \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat(\u001b[43mdf_markers_rank_significance_logreg\u001b[49m)\u001b[38;5;241m.\u001b[39mreset_index()\n\u001b[1;32m      6\u001b[0m df\u001b[38;5;241m.\u001b[39mcolumns \u001b[38;5;241m=\u001b[39m xenium_levels \u001b[38;5;241m+\u001b[39m df\u001b[38;5;241m.\u001b[39mcolumns[\u001b[38;5;28mlen\u001b[39m(xenium_levels) :]\u001b[38;5;241m.\u001b[39mtolist()\n\u001b[1;32m      7\u001b[0m df \u001b[38;5;241m=\u001b[39m df\u001b[38;5;241m.\u001b[39mquery(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcti == @cti and ctj == @ctj\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'df_markers_rank_significance_logreg' is not defined"
     ]
    }
   ],
   "source": [
    "cti = 'T cell'\n",
    "ctj = 'malignant cell'\n",
    "xenium_levels = [\"segmentation\", \"condition\", \"panel\", \"donor\", \"sample\",\"cti\",\"ctj\"]\n",
    "\n",
    "df = pd.concat(df_markers_rank_significance_logreg).reset_index()\n",
    "df.columns = xenium_levels + df.columns[len(xenium_levels) :].tolist()\n",
    "df = df.query(\"cti == @cti and ctj == @ctj\")\n",
    "df['-log10pvalue'] = -np.log10(df['hypergeometric_pvalue'])\n",
    "\n",
    "\n",
    "std_seurat_analysis_dir = Path(cfg['xenium_std_seurat_analysis_dir'])\n",
    "cell_type_annotation_dir = Path(cfg['xenium_cell_type_annotation_dir'])\n",
    "results_dir = Path(cfg['results_dir'])\n",
    "palette_dir = Path(cfg['xenium_metadata_dir'])\n",
    "segmentation_palette = palette_dir / 'col_palette_segmentation.csv'\n",
    "\n",
    "hue = \"segmentation\"\n",
    "hue_order = [\n",
    "    \"10x_mm_0um\",\n",
    "    \"10x_mm_5um\",\n",
    "    \"10x_mm_15um\",\n",
    "    \"10x_0um\",\n",
    "    \"10x_5um\",\n",
    "    \"10x_15um\",\n",
    "    \"baysor\",\n",
    "    \"proseg_expected\",\n",
    "    \"proseg_mode\",\n",
    "    \"segger\",\n",
    "]\n",
    "\n",
    "\n",
    "palette = pd.read_csv(segmentation_palette, index_col=0).iloc[:, 0]\n",
    "\n",
    "\n",
    "# plotting params, palette\n",
    "title = f\"Reference: {reference}, Method: {method}, Level: {level} \\n{cti} contaminated by {ctj}\"\n",
    "unique_labels = [c for c in hue_order if c in np.unique(df[hue].dropna())]\n",
    "unique_labels = unique_labels + [c for c in np.unique(df[hue].dropna()) if c not in unique_labels]\n",
    "palette = {u: palette[u] for u in unique_labels}\n",
    "legend_handles = [mpatches.Patch(color=color, label=label) for label, color in palette.items()]\n",
    "\n",
    "\n",
    "### hypergeometric pvalue boxplot\n",
    "f = plt.figure(figsize=(6, 6))\n",
    "ax = plt.subplot()\n",
    "g = sns.boxplot(df,x='panel',y='-log10pvalue', \n",
    "                hue=hue, hue_order=unique_labels, \n",
    "                legend=False, palette=palette,ax=ax\n",
    "                )\n",
    "\n",
    "plt.ylabel(r'$-\\log_{10} \\text{ p-value}$')\n",
    "sns.despine(offset=10, trim=True)\n",
    "ax.yaxis.grid(True)\n",
    "\n",
    "plt.suptitle(title)\n",
    "f.legend(\n",
    "    handles=legend_handles,\n",
    "    loc=\"center left\",\n",
    "    bbox_to_anchor=(1, 0.5),\n",
    "    title=hue,\n",
    "    frameon=False,\n",
    ")\n",
    "plt.tight_layout(rect=[0, 0, 1, 0.95])\n",
    "# plt.savefig(out_file, dpi=dpi, bbox_inches=\"tight\")\n",
    "plt.show()\n",
    "\n",
    "\n",
    "### NES boxplot\n",
    "f = plt.figure(figsize=(6, 6))\n",
    "ax = plt.subplot()\n",
    "g = sns.boxplot(df,x='panel',y='NES', \n",
    "                hue=hue, hue_order=unique_labels, \n",
    "                legend=False, palette=palette,ax=ax\n",
    "                )\n",
    "\n",
    "sns.despine(offset=10, trim=True)\n",
    "ax.yaxis.grid(True)\n",
    "\n",
    "plt.suptitle(title)\n",
    "f.legend(\n",
    "    handles=legend_handles,\n",
    "    loc=\"center left\",\n",
    "    bbox_to_anchor=(1, 0.5),\n",
    "    title=hue,\n",
    "    frameon=False,\n",
    ")\n",
    "plt.tight_layout(rect=[0, 0, 1, 0.95])\n",
    "# plt.savefig(out_file, dpi=dpi, bbox_inches=\"tight\")\n",
    "plt.show()\n",
    "\n",
    "\n",
    "### number of hits boxplot\n",
    "f = plt.figure(figsize=(6, 6))\n",
    "ax = plt.subplot()\n",
    "g = sns.boxplot(df,x='panel',y=f'n_hits_{top_n=}', \n",
    "                hue=hue, hue_order=unique_labels, \n",
    "                legend=False, palette=palette,ax=ax\n",
    "                )\n",
    "\n",
    "sns.despine(offset=10, trim=True)\n",
    "ax.yaxis.grid(True)\n",
    "\n",
    "plt.suptitle(title)\n",
    "f.legend(\n",
    "    handles=legend_handles,\n",
    "    loc=\"center left\",\n",
    "    bbox_to_anchor=(1, 0.5),\n",
    "    title=hue,\n",
    "    frameon=False,\n",
    ")\n",
    "plt.tight_layout(rect=[0, 0, 1, 0.95])\n",
    "# plt.savefig(out_file, dpi=dpi, bbox_inches=\"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "sns.set_style('ticks')\n",
    "ref_segmentation = '10x_5um'\n",
    "\n",
    "df = pd.concat(df_markers_rank_significance_diffexpr).reset_index()\n",
    "df['-log10pvalue'] = -np.log10(df['hypergeometric_pvalue'])\n",
    "df.columns = xenium_levels + df.columns[len(xenium_levels) :].tolist()\n",
    "u_condition_panel = df[['condition','panel']].drop_duplicates().values\n",
    "\n",
    "metrics = ['NES', '-log10pvalue', f'n_hits_{top_n=}']\n",
    "\n",
    "for metric in metrics:\n",
    "    for condition,panel in u_condition_panel:\n",
    "\n",
    "        df_plot = df.query(f\"segmentation == '{ref_segmentation}' and condition == '{condition}' and panel == '{panel}'\")\n",
    "        df_plot = df_plot.groupby(['cti', 'ctj'])[metric].mean().unstack()\n",
    "        df_plot = df_plot.loc[df_plot.sum(1).sort_values(ascending=False).index]\n",
    "        df_plot = df_plot[df_plot.sum(0).sort_values(ascending=False).index]\n",
    "\n",
    "        f = plt.figure(figsize=(8,8))\n",
    "        ax = plt.subplot()\n",
    "        ax.set_title(f\"{condition=} {panel=} {metric=}\",fontsize=20)\n",
    "        g = sns.heatmap(df_plot,cmap='coolwarm',center=-np.log10(0.05) if metric == '-log10pvalue' else 0.)\n",
    "        plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cti = 'T cell'\n",
    "ctj = 'malignant cell'\n",
    "xenium_levels = [\"segmentation\", \"condition\", \"panel\", \"donor\", \"sample\",\"cti\",\"ctj\"]\n",
    "\n",
    "df = pd.concat(df_markers_rank_significance_logreg).reset_index()\n",
    "df.columns = xenium_levels + df.columns[len(xenium_levels) :].tolist()\n",
    "df = df.query(\"cti == @cti and ctj == @ctj\")\n",
    "df['-log10pvalue'] = -np.log10(df['hypergeometric_pvalue'])\n",
    "\n",
    "\n",
    "### hypergeometric pvalue boxplot\n",
    "f = plt.figure(figsize=(6, 6))\n",
    "ax = plt.subplot()\n",
    "g = sns.boxplot(df,x='panel',y='-log10pvalue', \n",
    "                hue=hue, hue_order=unique_labels, \n",
    "                legend=False, palette=palette,ax=ax\n",
    "                )\n",
    "\n",
    "plt.ylabel(r'$-\\log_{10} \\text{ p-value}$')\n",
    "sns.despine(offset=10, trim=True)\n",
    "ax.yaxis.grid(True)\n",
    "\n",
    "plt.suptitle(title)\n",
    "f.legend(\n",
    "    handles=legend_handles,\n",
    "    loc=\"center left\",\n",
    "    bbox_to_anchor=(1, 0.5),\n",
    "    title=hue,\n",
    "    frameon=False,\n",
    ")\n",
    "plt.tight_layout(rect=[0, 0, 1, 0.95])\n",
    "# plt.savefig(out_file, dpi=dpi, bbox_inches=\"tight\")\n",
    "plt.show()\n",
    "\n",
    "\n",
    "### NES boxplot\n",
    "f = plt.figure(figsize=(6, 6))\n",
    "ax = plt.subplot()\n",
    "g = sns.boxplot(df,x='panel',y='NES', \n",
    "                hue=hue, hue_order=unique_labels, \n",
    "                legend=False, palette=palette,ax=ax\n",
    "                )\n",
    "\n",
    "sns.despine(offset=10, trim=True)\n",
    "ax.yaxis.grid(True)\n",
    "\n",
    "plt.suptitle(title)\n",
    "f.legend(\n",
    "    handles=legend_handles,\n",
    "    loc=\"center left\",\n",
    "    bbox_to_anchor=(1, 0.5),\n",
    "    title=hue,\n",
    "    frameon=False,\n",
    ")\n",
    "plt.tight_layout(rect=[0, 0, 1, 0.95])\n",
    "# plt.savefig(out_file, dpi=dpi, bbox_inches=\"tight\")\n",
    "plt.show()\n",
    "\n",
    "\n",
    "### number of hits boxplot\n",
    "f = plt.figure(figsize=(6, 6))\n",
    "ax = plt.subplot()\n",
    "g = sns.boxplot(df,x='panel',y=f'n_hits_{top_n=}', \n",
    "                hue=hue, hue_order=unique_labels, \n",
    "                legend=False, palette=palette,ax=ax\n",
    "                )\n",
    "\n",
    "sns.despine(offset=10, trim=True)\n",
    "ax.yaxis.grid(True)\n",
    "\n",
    "plt.suptitle(title)\n",
    "f.legend(\n",
    "    handles=legend_handles,\n",
    "    loc=\"center left\",\n",
    "    bbox_to_anchor=(1, 0.5),\n",
    "    title=hue,\n",
    "    frameon=False,\n",
    ")\n",
    "plt.tight_layout(rect=[0, 0, 1, 0.95])\n",
    "# plt.savefig(out_file, dpi=dpi, bbox_inches=\"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.concat(df_permutations_logreg).reset_index()\n",
    "df.columns = xenium_levels + df.columns[len(xenium_levels) :].tolist()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "spatial",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
