{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/jbac/miniforge3/envs/spatial/lib/python3.11/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
      "  from .autonotebook import tqdm as notebook_tqdm\n"
     ]
    }
   ],
   "source": [
    "from pathlib import Path\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import ovrlpy\n",
    "\n",
    "import sys\n",
    "sys.path.extend(['../../scripts','../../scripts/xenium'])\n",
    "import readwrite\n",
    "cfg = readwrite.config()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running vertical adjustment\n",
      "Creating gene expression embeddings for visualization:\n",
      "Analyzing in 3d mode:\n",
      "determining pseudocells:\n",
      "found 11537 pseudocells\n",
      "sampling expression:\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DMKN:  28%|██▊       | 77/280 [01:13<02:30,  1.35it/s]    "
     ]
    }
   ],
   "source": [
    "path = Path('/work/PRTNR/CHUV/DIR/rgottar1/spatial/env/xenium_paper/data/xenium/processed/segmentation/10x_0um/breast/breast/1256/1256/normalised_results/outs/transcripts.parquet')\n",
    "coordinate_df = pd.read_parquet(path).query(\"is_gene & (qv >= 20)\") \\\n",
    "    .rename(\n",
    "        columns={\n",
    "            \"x_location\": \"x\",\n",
    "            \"y_location\": \"y\",\n",
    "            \"z_location\": \"z\",\n",
    "            \"feature_name\": \"gene\",\n",
    "        }) \\\n",
    "    .query(\"is_gene & (qv >= 20)\")  # remove dummy molecules\n",
    "\n",
    "coordinate_df[\"gene\"] = coordinate_df[\"gene\"].astype(\"category\")\n",
    "\n",
    "# coordinate_df[::1000].plot.scatter(x=\"x\", y=\"y\", s=1)\n",
    "# coordinate_df = coordinate_df.copy()  # make a copy to avoid SettingWithCopyWarning\n",
    "\n",
    "signal_integrity, signal_strength, visualizer = ovrlpy.run(\n",
    "    df=coordinate_df, cell_diameter=10, n_expected_celltypes=20\n",
    ")\n",
    "\n",
    "fig, ax = ovrlpy.plot_signal_integrity(\n",
    "    signal_integrity, signal_strength, signal_threshold=3\n",
    ")\n",
    "\n",
    "doublet_df = ovrlpy.detect_doublets(\n",
    "    signal_integrity, signal_strength, minimum_signal_strength=3, integrity_sigma=2\n",
    ")\n",
    "\n",
    "_ = plt.scatter(\n",
    "    doublet_df[\"x\"],\n",
    "    doublet_df[\"y\"],\n",
    "    c=doublet_df[\"integrity\"],\n",
    "    s=1,\n",
    "    cmap=\"viridis\",\n",
    "    vmin=0,\n",
    "    vmax=1,\n",
    ")\n",
    "plt.gca().set_aspect(\"equal\")\n",
    "plt.colorbar()\n",
    "\n",
    "\n",
    "window_size = 60\n",
    "doublet_case = 0\n",
    "\n",
    "x, y = doublet_df.loc[doublet_case, [\"x\", \"y\"]]\n",
    "\n",
    "ovrlpy.plot_region_of_interest(\n",
    "    x,\n",
    "    y,\n",
    "    coordinate_df,\n",
    "    visualizer,\n",
    "    signal_integrity,\n",
    "    signal_strength,\n",
    "    window_size=window_size,\n",
    ")\n",
    "\n",
    "\n",
    "# visualizer.save(result_folder / \"Xenium-text-visualizer.vis\")\n",
    "\n",
    "# vis = ovrlpy.load_visualizer(result_folder / \"Xenium-text-visualizer.vis\")\n",
    "\n",
    "plt.figure(figsize=(20, 5))\n",
    "\n",
    "ax = plt.subplot(111, projection=\"3d\")\n",
    "\n",
    "for i in range(-2, 3):\n",
    "    subset = coordinate_df[(coordinate_df.z - coordinate_df.z_delim).between(i, i + 1)]\n",
    "\n",
    "    ax.scatter(\n",
    "        subset.x[::100],\n",
    "        subset.y[::100],\n",
    "        np.zeros(1 + (len(subset) // 100)) + i,\n",
    "        s=1,\n",
    "        alpha=0.1,\n",
    "    )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_947089/2333320809.py:3: FutureWarning: The default value of observed=False is deprecated and will change to observed=True in a future version of pandas. Specify observed=False to silence this warning and retain the current behavior\n",
      "  cm = transcripts.pivot_table(index=cell_column, columns=feature_column, aggfunc='size', fill_value=0)\n"
     ]
    }
   ],
   "source": [
    "def transcripts_to_count_matrix(transcripts,cell_column='cell_id',feature_column='feature_name', qv_treshold=20):\n",
    "    transcripts = transcripts.query(f\"(qv >= {qv_treshold}) & {cell_column} != 'UNASSIGNED'\")\n",
    "    cm = transcripts.pivot_table(index=cell_column, columns=feature_column, aggfunc='size', fill_value=0)\n",
    "    return cm\n",
    "\n",
    "integrity_filtering_threshold = 0.5\n",
    "\n",
    " # x and y are transposed in ovrlpy output. sanity check x and y must be transposed\n",
    "assert (coordinate_df.y_pixel.max()+1) == signal_integrity.shape[0]\n",
    "assert (coordinate_df.x_pixel.max()+1) == signal_integrity.shape[1]\n",
    "\n",
    "coordinate_df['signal_integrity'] = signal_integrity[coordinate_df.y_pixel,coordinate_df.x_pixel]\n",
    "coordinate_df['signal_strength'] = signal_strength[coordinate_df.y_pixel,coordinate_df.x_pixel]\n",
    "\n",
    "coordinate_df_filtered = coordinate_df[coordinate_df.signal_integrity > integrity_filtering_threshold]\n",
    "cell_mean_integrity = coordinate_df.query(\"cell_id != 'UNASSIGNED'\").groupby('cell_id')['signal_integrity'].mean()\n",
    "cell_is_singlet = cell_mean_integrity > integrity_filtering_threshold\n",
    "\n",
    "count_matrix_filtered = transcripts_to_count_matrix(coordinate_df_filtered,feature_column='gene')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>signal_integrity</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cell_id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>aaaabihh-1</th>\n",
       "      <td>0.228764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aaabddfm-1</th>\n",
       "      <td>0.160387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aaabeakn-1</th>\n",
       "      <td>0.657175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aaachkkh-1</th>\n",
       "      <td>0.763619</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>aaadbefk-1</th>\n",
       "      <td>0.703907</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>oimeelcd-1</th>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>oimefhce-1</th>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>oimefmmc-1</th>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>oimfkedc-1</th>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>oimflfjp-1</th>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>140511 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            signal_integrity\n",
       "cell_id                     \n",
       "aaaabihh-1          0.228764\n",
       "aaabddfm-1          0.160387\n",
       "aaabeakn-1          0.657175\n",
       "aaachkkh-1          0.763619\n",
       "aaadbefk-1          0.703907\n",
       "...                      ...\n",
       "oimeelcd-1          0.000000\n",
       "oimefhce-1          0.000000\n",
       "oimefmmc-1          0.000000\n",
       "oimfkedc-1          0.000000\n",
       "oimflfjp-1          0.000000\n",
       "\n",
       "[140511 rows x 1 columns]"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cell_mean_integrity.to_frame()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABCC11</th>\n",
       "      <td>ABCC11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTA2</th>\n",
       "      <td>ACTA2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTG2</th>\n",
       "      <td>ACTG2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ADAM9</th>\n",
       "      <td>ADAM9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ADGRE5</th>\n",
       "      <td>ADGRE5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VWF</th>\n",
       "      <td>VWF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WARS</th>\n",
       "      <td>WARS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZEB1</th>\n",
       "      <td>ZEB1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZEB2</th>\n",
       "      <td>ZEB2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZNF562</th>\n",
       "      <td>ZNF562</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>280 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          gene\n",
       "gene          \n",
       "ABCC11  ABCC11\n",
       "ACTA2    ACTA2\n",
       "ACTG2    ACTG2\n",
       "ADAM9    ADAM9\n",
       "ADGRE5  ADGRE5\n",
       "...        ...\n",
       "VWF        VWF\n",
       "WARS      WARS\n",
       "ZEB1      ZEB1\n",
       "ZEB2      ZEB2\n",
       "ZNF562  ZNF562\n",
       "\n",
       "[280 rows x 1 columns]"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "spatial",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
